src/main/java/com/mongodb/spark/sql/connector/read/partitioner/FieldPartitioner.java

```diff
@@ -82,44 +79,31 @@
 
               BsonDocument matchFilter = new BsonDocument();
               if (previous != null) {
-                partitionFieldList.forEach(
-                    k -> {
-                      if (previous.containsKey(k)) {
-                        matchFilter.put(k, new BsonDocument("$gte", previous.get(k)));
-                      }
-                    });
+                matchFilter.put(
+                    partitionField, new BsonDocument("$gte", previous.get(partitionField)));
               }
 
               if (current != null) {
-                partitionFieldList.forEach(
-                    k -> {
-                      if (current.containsKey(k)) {
-                        matchFilter.put(
-                            k,
-                            matchFilter
-                                .getDocument(k, new BsonDocument())
-                                .append("$lt", current.get(k)));
-                      }
-                    });
+                matchFilter.put(
+                    partitionField,
+                    matchFilter
+                        .getDocument(partitionField, new BsonDocument())
+                        .append("$lt", current.get(partitionField)));
               }
 
               if (previous != null && current != null) {
-                for (String k : partitionFieldList) {
-                  int comparision =
-                      BSON_VALUE_COMPARATOR.compare(
-                          current.get(k, BsonNull.VALUE), previous.get(k, BsonNull.VALUE));
-                  if (comparision < 0) {
-                    throw new ConfigException(
-                        "Invalid partitioner configuration. "
-                            + "The partitions generated should be contingous and the partition values should be ascending in "
-                            + "the partitions to ensure no duplicated data.");
-                  } else if (comparision == 0) {
-                    throw new ConfigException(
-                        format(
-                            "Invalid partitioner configuration. The partitions generated contain duplicates "
-                                + "for the field: `%s`",
-                            k));
-                  }
+                int comparision = BSON_VALUE_COMPARATOR.compare(current, previous);
+                if (comparision < 0) {
+                  throw new ConfigException(
+                      "Invalid partitioner configuration. "
+                          + "The partitions generated should be contiguous and the partition values should be ascending in "
+                          + "the partitions to ensure no duplicated data.");
```

Comment:
Since `createMongoInputPartitions` is package-access, custom `FieldPartitioner` implementations cannot call this method. Consequently, `comparison < 0` cannot be true regardless of the user data, unless our own implementations of `FieldPartitioner` are incorrect, in which case we must not throw a `RuntimeException` (we still may throw an `AssertionError`).

Reply:
I think I've now caught up / caught on apologies for being slow! 

Given that we use MongoDB sorting in our implementations. We can then use `new HashSet(upperBounds).size() != upperbounds.size()` to determine if there are any duplicates and that negates the whole reason for doing any comparisons and `BsonValueComparator` can be removed.